ports/nrf/Makefile

```diff
@@ -128,7 +129,15 @@ CFLAGS_MCU_m4 = $(CFLAGS_CORTEX_M) -mtune=cortex-m4 -mcpu=cortex-m4 -mfpu=fpv4-s
 
 CFLAGS_MCU_m0 = $(CFLAGS_CORTEX_M) -fshort-enums -mtune=cortex-m0 -mcpu=cortex-m0 -mfloat-abi=soft
 
+# linker wrap does not work with lto on older gcc/binutils: https://sourceware.org/bugzilla/show_bug.cgi?id=24406
+GCC_VERSION = $(shell arm-none-eabi-gcc --version | sed -n -E 's:^arm.*([0-9]+\.[0-9]+\.[0-9]+).*$$:\1:p')
+GCC_MAJOR_VERS = $(word 1,$(subst ., ,$(GCC_VERSION)))
+ifeq ($(shell test $(GCC_MAJOR_VERS) -ge 10; echo $$?),0)
```

Comment:
@andrewleech this looks a bit funny to me.  On my Arch system I have:
```
arm-none-eabi-gcc (Arch Repository) 14.1.0
```
But `GCC_VERSION` here evaluates to `4.1.0`.

Reply:
Ah drats, it worked on a few systems I'd tested on but yes my arch server is also at 14.1.0 and I get the same thing. Silly greedy regex. Easy fix...
`arm-none-eabi-gcc --version | sed -n -E 's:^arm.*([0-9]+\.[0-9]+\.[0-9]+).*$$:\1:p'` ->  
`arm-none-eabi-gcc --version | sed -n -E 's:^arm.* ([0-9]+\.[0-9]+\.[0-9]+).*$$:\1:p'`

Reply:
https://github.com/micropython/micropython/pull/15553

I wish there was a way to do this in a proper programatic way, but I couldn't find one.